### PR TITLE
Js playcardbugs+militaryscores+newfrontendinfo

### PR DESCRIPTION
**Pull request by jacobstringer**
Fixes several bugs around playing cards - I got it to play into the second age without back end issues now.

Includes all players info for front end (but only your hand), along with maximum slots for wonders and research card counters.

Includes military point calculations at the end of each round.

Reinstates polling.

Includes guards to stop multiple card playing per round, and avoids multiple players being created or hands being dealt due to weird latency bugs that I was experiencing.

----------------------

Introduces some issues with the front end, presumably due to the change in information sent through:
- Wonder board not showing now
- Goes back to 'start game' screen each time, which needs pushing to send back through to game screen

I am not sure what is causing this but I am guessing it is the updated JSON object being sent. Here is an example of what is sent to the back (array of players, then array of cards for the one player) from print_json... doesn't include a few extras that are added in some API calls which have remained unchanged:

{'player': [{'id': 1, 'gameId': 1, 'userId': 1, 'ready': True, 'left_id': 3, 'right_id': 2, 'wonder': 'The Hanging Gardens of Babylon', 'wonder_level': 0, 'max_wonder': 3, 'wood': 0, 'brick': 1, 'ore': 0, 'stone': 0, 'glass': 0, 'paper': 0, 'cloth': 0, 'extra_wood': 0, 'extra_brick': 0, 'extra_ore': 0, 'extra_stone': 0, 'extra_glass': 0, 'extra_paper': 0, 'extra_cloth': 0, 'points': 0, 'military': 0, 'money': 3, 'cog': 0, 'tablet': 0, 'compass': 0}, {'id': 2, 'gameId': 1, 'userId': 4, 'ready': True, 'left_id': 1, 'right_id': 3, 'wonder': 'The Lighthouse of Alexandria', 'wonder_level': 0, 'max_wonder': 3, 'wood': 0, 'brick': 0, 'ore': 0, 'stone': 0, 'glass': 1, 'paper': 0, 'cloth': 0, 'extra_wood': 0, 'extra_brick': 0, 'extra_ore': 0, 'extra_stone': 0, 'extra_glass': 0, 'extra_paper': 0, 'extra_cloth': 0, 'points': 0, 'military': 0, 'money': 3, 'cog': 0, 'tablet': 0, 'compass': 0}, {'id': 3, 'gameId': 1, 'userId': 2, 'ready': True, 'left_id': 2, 'right_id': 1, 'wonder': 'The Mausoleum of Halicarnassus', 'wonder_level': 0, 'max_wonder': 3, 'wood': 0, 'brick': 0, 'ore': 0, 'stone': 0, 'glass': 0, 'paper': 0, 'cloth': 1, 'extra_wood': 0, 'extra_brick': 0, 'extra_ore': 0, 'extra_stone': 0, 'extra_glass': 0, 'extra_paper': 0, 'extra_cloth': 0, 'points': 0, 'military': 0, 'money': 3, 'cog': 0, 'tablet': 0, 'compass': 0}],'cards': [{'id': 29, 'name': 'Lumber Yard', 'noPlayers': 3, 'age': 1, 'colour': 'brown', 'costWood': 0, 'costBrick': 0, 'costOre': 0, 'costStone': 0, 'costGlass': 0, 'costPaper': 0, 'costCloth': 0, 'prerequisite1': '', 'prerequisite2': '', 'giveWood': 1, 'giveBrick': 0, 'giveOre': 0, 'giveStone': 0, 'giveGlass': 0, 'givePaper': 0, 'giveCloth': 0, 'resourceAlternating': False, 'givePoints': 0, 'giveMilitary': 0, 'giveMoney': 0, 'giveResearch': ''}, {'id': 33, 'name': 'Timber Yard', 'noPlayers': 3, 'age': 1, 'colour': 'brown', 'costWood': 0, 'costBrick': 0, 'costOre': 0, 'costStone': 0, 'costGlass': 0, 'costPaper': 0, 'costCloth': 0, 'prerequisite1': '', 'prerequisite2': '', 'giveWood': 1, 'giveBrick': 0, 'giveOre': 0, 'giveStone': 1, 'giveGlass': 0, 'givePaper': 0, 'giveCloth': 0, 'resourceAlternating': True, 'givePoints': 0, 'giveMilitary': 0, 'giveMoney': 0, 'giveResearch': ''}, {'id': 40, 'name': 'Marketplace', 'noPlayers': 3, 'age': 1, 'colour': 'yellow', 'costWood': 0, 'costBrick': 0, 'costOre': 0, 'costStone': 0, 'costGlass': 0, 'costPaper': 0, 'costCloth': 0, 'prerequisite1': '', 'prerequisite2': '', 'giveWood': 0, 'giveBrick': 0, 'giveOre': 0, 'giveStone': 0, 'giveGlass': 0, 'givePaper': 0, 'giveCloth': 0, 'resourceAlternating': False, 'givePoints': 0, 'giveMilitary': 0, 'giveMoney': 0, 'giveResearch': ''}, {'id': 42, 'name': 'Theatre', 'noPlayers': 3, 'age': 1, 'colour': 'blue', 'costWood': 0, 'costBrick': 0, 'costOre': 0, 'costStone': 0, 'costGlass': 0, 'costPaper': 0, 'costCloth': 0, 'prerequisite1': '', 'prerequisite2': '', 'giveWood': 0, 'giveBrick': 0, 'giveOre': 0, 'giveStone': 0, 'giveGlass': 0, 'givePaper': 0, 'giveCloth': 0, 'resourceAlternating': False, 'givePoints': 2, 'giveMilitary': 0, 'giveMoney': 0, 'giveResearch': ''}, {'id': 43, 'name': 'Baths', 'noPlayers': 3, 'age': 1, 'colour': 'blue', 'costWood': 0, 'costBrick': 0, 'costOre': 0, 'costStone': 1, 'costGlass': 0, 'costPaper': 0, 'costCloth': 0, 'prerequisite1': '', 'prerequisite2': '', 'giveWood': 0, 'giveBrick': 0, 'giveOre': 0, 'giveStone': 0, 'giveGlass': 0, 'givePaper': 0, 'giveCloth': 0, 'resourceAlternating': False, 'givePoints': 3, 'giveMilitary': 0, 'giveMoney': 0, 'giveResearch': ''}, {'id': 47, 'name': 'Apothecary', 'noPlayers': 3, 'age': 1, 'colour': 'green', 'costWood': 0, 'costBrick': 0, 'costOre': 0, 'costStone': 0, 'costGlass': 0, 'costPaper': 0, 'costCloth': 1, 'prerequisite1': '', 'prerequisite2': '', 'giveWood': 0, 'giveBrick': 0, 'giveOre': 0, 'giveStone': 0, 'giveGlass': 0, 'givePaper': 0, 'giveCloth': 0, 'resourceAlternating': False, 'givePoints': 0, 'giveMilitary': 0, 'giveMoney': 0, 'giveResearch': 'compass'}, {'id': 48, 'name': 'Workshop', 'noPlayers': 3, 'age': 1, 'colour': 'green', 'costWood': 0, 'costBrick': 0, 'costOre': 0, 'costStone': 0, 'costGlass': 1, 'costPaper': 0, 'costCloth': 0, 'prerequisite1': '', 'prerequisite2': '', 'giveWood': 0, 'giveBrick': 0, 'giveOre': 0, 'giveStone': 0, 'giveGlass': 0, 'givePaper': 0, 'giveCloth': 0, 'resourceAlternating': False, 'givePoints': 0, 'giveMilitary': 0, 'giveMoney': 0, 'giveResearch': 'cog'}]}

---------------------------------------------

Update - an additional player field has been added, called 'military_loss' for front end use.

Fixed a couple of testing bits and pieces that I didn't get rid of earlier.

### CODE REVIEW

File: application/controllers/state.py [hunk 1]
```diff
@@ -2,8 +2,21 @@
...
+        'cards': [card.serialise() for card in cards],
+    })"""
     return {
-        'player': player.serialise(),
+        'player': [player.serialise() for player in players],
         'cards': [card.serialise() for card in cards],
     }
+
+
+def false_true (string):
```
Comment (mmdonaldson) [inline application/controllers/state.py:17]: I think we should have a more explicit function name here

**Review by mmdonaldson — CHANGES_REQUESTED** [review]
if you could please just update the function I commented on earlier it's good to go

**Review by mmdonaldson — APPROVED** [review]

**Comment by jacobstringer** [comment]
[Return extra data from game api response](https://trello.com/c/SaQ9T7OZ/44-return-extra-data-from-game-api-response)

**Comment by jacobstringer** [comment]
[Return extra data from game api response](https://trello.com/c/SaQ9T7OZ/44-return-extra-data-from-game-api-response)

**Comment by jacobstringer** [comment]
[End of age war calculations](https://trello.com/c/Y9Vm0SEK/34-end-of-age-war-calculations)

**Pull request merged**